sympy/sets/sets.py

```diff
@@ -799,6 +799,37 @@ def Ropen(cls, a, b):
         """Return an interval not including the right boundary."""
         return cls(a, b, False, True)
 
+    @classmethod
+    def map(cls, i, ii, x = None):
+        """
+        Return an expression that maps the first interval
+        into the second.
+
+        Examples
+        ========
+
+        >> from sympy import Interval
+        >> from sympy.abc import a, b, x
+        >> from sympy import factor
+        >> e = Interval.map(Interval(10, 12), Interval(-1, 1))
+        >> print(e.subs(x, 10))
+        0
```

Comment:
Hi,

actually, there was a typo. 10 maps to -1, 11 to 0 and 12 to 1.
I fixed it in my latest branch.

Also, I made x a required argument, and added
examples/beginner/map_intervals.py


2015-11-16 22:30 GMT+01:00 Aaron Meurer <notifications@github.com>:

> In sympy/sets/sets.py
> <https://github.com/sympy/sympy/pull/10144#discussion_r44986094>:
>
> > @@ -799,6 +799,37 @@ def Ropen(cls, a, b):
> >          """Return an interval not including the right boundary."""
> >          return cls(a, b, False, True)
> >
> > +    @classmethod
> > +    def map(cls, i, ii, x = None):
> > +        """
> > +        Return an expression that maps the first interval
> > +        into the second.
> > +
> > +        Examples
> > +        ========
> > +
> > +        >> from sympy import Interval
> > +        >> from sympy.abc import a, b, x
> > +        >> from sympy import factor
> > +        >> e = Interval.map(Interval(10, 12), Interval(-1, 1))
> > +        >> print(e.subs(x, 10))
> > +        0
>
> Shouldn't 10 map to -1 and 12 map to 1?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/10144/files#r44986094>.
>